src/features/MigrateLS/MigrateLS.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo, useEffect } from 'react';
+import { useReducer } from 'reinspect';
+import styled from 'styled-components';
+
+import { IFrame, Downloader, Link } from '@components';
+import MigrateLSReducer, { defaultState, UIStates } from './reducer';
+import { bindActions } from './actions';
+import { getIFrameSrc } from './helpers';
+
+const SActionContainer = styled.div`
+  display: inline-block;
+  margin: 0 1ch;
+  & button {
+    padding: 1px 6px;
+  }
+  & button:last-of-type {
+    margin-left: 0.5ch;
+  }
+`;
+
+const SBanner = styled.div`
+  height: 3em;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+`;
+
+/**
+ *  Use an IFrame to check if the old url contains a localstorage that we need to import.
+ *  Validate the schema, and prompt the user to import their previous settings.
+ *  If user accepts and migratation succeds:
+ *    - destroy the previous storage so they aren't prompted again.
+ *    - let `NO_ACCOUNTS` handle the redirect, after `Store` is hydrated.
+ *  If user cacels:
+ *    - prompt to confirm cancel and inform that it will download a json
+ *    - destroy previous storage.
+ */
+const MigrateLS = ({
+  isDefault: isDefaultStore = false,
+  importStorage = () => false,
+  isValidImport
+}: {
+  isDefault?: boolean;
+  importStorage(ls: string): boolean;
+  isValidImport(ls: string): boolean;
+}) => {
+  // We only need to get the iFrame src url once
+  const src = useMemo(() => getIFrameSrc(window), []);
+
+  const [{ storage, iframeRef, uiState, canDestroy, canDownload, error }, dispatch] = useReducer(
+    MigrateLSReducer,
+    defaultState,
+    (state) => state,
+    'MigrateLS'
+  );
+
+  // Connect our actions to the store
+  const {
+    getStorage,
+    destroyStorage,
+    downloadAndDestroy,
+    cancelMigration,
+    migrateStorage,
+    abortCancel
+  } = useMemo(() => bindActions(dispatch), [dispatch]);
+
+  useEffect(() => {
+    if (canDestroy && iframeRef) {
+      destroyStorage(iframeRef);
+    }
+  }, [canDestroy, canDownload]);
+
+  const handleLoad = (frame: HTMLIFrameElement) => getStorage(frame, src, isValidImport);
+  const handleCancel = () => cancelMigration();
+  const handleImport = () => {
+    if (!storage) return;
+    migrateStorage(storage, importStorage);
+  };
+  const handleConfirm = () => {
+    downloadAndDestroy();
+  };
+  const handleAbort = () => abortCancel();
+
+  const UI_STATES: Record<UIStates, JSX.Element> = {
+    default: <></>,
+    'migrate-prompt': (
+      <SBanner>
+        We found your previous settings from beta.mycrypto.com. Would you like to import them?
```

Comment:
Use translation strings for all of this.

Reply:
This feature will be deleted before we get translations.
I added a card so we remember to sunset it.
https://app.clubhouse.io/mycrypto/story/6687/sunset-migratels